### PR TITLE
Added aria-label to boxes to popup and options html

### LINKED ISSUE #125: Improve accessibility of messages
Problem: When you do not see the icons/style of the messages, you cannot differentiate whether these are errors, info messages, or so…

Solution (at least what I'd say is the easiest/best): Add `aria-label` to the message icon (or box?).

Of course, the static HTML needs to be adjusted.
In the `MessageHandler` you need to adjust the `setMessageDesign` and the corresponding tests, to also check this attribute.

### PR DESCRIPTION
**Pull request by RShalman**
Issue #125 
Now I've added labels for live code.
Let me know if it's what you were asking for)

P.S. I didn't quite get what you wanted to adjust in <code>setMessageDesign</code> function (better say, how to adjust it).

**Edit (by @rugk)**: Fixes #125

### CODE REVIEW

**Comment by rugk** [comment]
BTW unfortunately adding the issue number to the title does not quite work on GitHub, you have to add it to the body and [have to follow the format defined here](https://help.github.com/articles/closing-issues-using-keywords/), i.e. "Fixes https://github.com/rugk/offline-qr-code/issues/125" would e.g. work.
(You can also edit your PR description on GitHub.)

Also notice you can just push to an existing branches (and thereby PRs). You do not have to open a new Pull Request for that.

Good that you found the live versions of the HTML files, these are what I meant, but that does not mean they should not be in the tests. So please include the changes [of your old PR](https://github.com/rugk/offline-qr-code/pull/130) also here (but with the short labels, obviously), i.e also adjust the test HTML files.

**Comment by rugk** [comment]
So I've added some tests for the `setMessageDesign` function. They should make clear what to adjust. (Also improved the function's description as it was a little off.)

File: src/options/options.html [hunk 1]
```diff
@@ -12,42 +12,42 @@
 
 	<body>
 		<div class="message-container">
-			<div id="messageTips" class="message-box info invisible fade-hide">
+			<div id="messageTips" aria-label="info" class="message-box info invisible fade-hide">
```
Comment (rugk) [inline src/options/options.html]:
```suggestion
			<div id="messageTips" aria-label="message tips" class="message-box info invisible fade-hide">
```

Remember the aria label should not specify the style/class, but useful for one who cannot see the visual styles. So I'd suggest `message tips` or `info box: message tips` or something like this.

File: src/popup/qrcode.html [hunk 2]
```diff
@@ -14,7 +14,7 @@
 	</head>
 	<body class="center-container">
 		<div id="topMessageContainer" class="message-container center-container">
-			<div id="messageTips" class="message-box info invisible fade-hide">
+			<div id="messageTips" aria-label="info" class="message-box info invisible fade-hide">
```
Comment (rugk) [inline src/popup/qrcode.html]:
```suggestion
			<div id="messageTips" aria-label="message tips" class="message-box info invisible fade-hide">
```

same here

File: src/popup/qrcode.html [hunk 3]
```diff
@@ -33,15 +33,15 @@
...
 						<span class="message-text"></span>
 						<img class="icon-dismiss invisible" src="/common/img/close.svg" width="24" height="24" tabindex="0" data-i18n data-i18n-aria-label="__MSG_dismissIconDescription__"></span>
 					</div>
-					<div id="messageError" class="message-box error flot-qr float-qr-bottom invisible fade-hide">
+					<div id="messageError" aria-label="error" class="message-box error flot-qr float-qr-bottom invisible fade-hide">
 						<span class="message-text" data-i18n="__MSG_couldNotGenerateQrCode__">Could not generate QR code.</span>
 						<img class="icon-dismiss invisible" src="/common/img/close-white.svg" width="24" height="24" tabindex="0" data-i18n data-i18n-aria-label="__MSG_dismissIconDescription__"></span>
 					</div>
-					<div id="messageLoading" class="message-box info flot-qr float-qr-bottom">
+					<div id="messageLoading" aria-label="info" class="message-box info flot-qr float-qr-bottom">
```
Comment (rugk) [inline src/popup/qrcode.html]:
```suggestion
					<div id="messageLoading" aria-label="loading" class="message-box info flot-qr float-qr-bottom">
```

The visual style is "info", but I would be fine with proving more info here, so "info box: loading" (or the thing I've adjusted above) may be an idea.

**Comment by rugk** [comment]
I just thought about the labels again and I think changing the aria text to `info box`, `success box` or `info message`, `success message` or so instead of just `info` , `success` may be more appropriate.
If you want to adjust that, feel free to do so. You also need to adjust the labels in the unit tests then. However, I can also do so, so do not feel obliged to adjust this now.

The small JS adjustment is the only thing missing from this PR.

**Comment by RShalman** [comment]
I think I'll do so by tomorrow. No problem =)

**Comment by rugk** [comment]
Great, so just FYI I'll want to release v1.3 of this add-on this weekend.

**Comment by rugk** [comment]
Great, so just FYI I'll want to release v1.3 of this add-on this weekend.

**Comment by RShalman** [comment]
Hi, mate! Sorry for being a bit late (not for the release, gladly!). Could you check my last changes if they're what you were asking for?

**Comment by rugk** [comment]
Sorry for being so late to reply. Also i have postponed the release a few days anyway. So I had a look at the code and the HTML code looks fine. However, as for the JS code your changes in the unit test files did not work in any way.

So the JS change is still outstanding. Note what i want:
* do not change the unit tests, I already adjusted them so they should succeed if you adjust the JS correctly.
* Furthermore it's not enough to just edit it on GitHub. You need to clone this repo to run the unit tests locally. However, it's really easy: See the ["Getting started"](https://github.com/rugk/offline-qr-code/blob/master/CONTRIBUTING.md#getting-started) section and the [second about how to run the unit tests](https://github.com/rugk/offline-qr-code/blob/master/CONTRIBUTING.md#tests).
  E.g. in your last commit you changed the unit tests by restting the HTML code that it is about to change. This did not help to bring the unit tests to a working state. E.g. you assigned values to a constants (`const`), which is even just syntactical invalid and would have been obvious if you run the code at least once.
* Note I already pointed you to the code you need to adjust [here](https://github.com/rugk/offline-qr-code/pull/131#pullrequestreview-166793973). (as for a direct link [here you go](https://github.com/rugk/offline-qr-code/blob/ab568b82dad12525be976c487be5c26fa0e2fa25/src/common/modules/MessageHandler.js#L263))

To get an overview again: The function that need adjustments **adjusts** a message's HTML code, so it does have another type/design, e.g. it changes the error message design to an info design. Obviously, it also has to change the `aria-label` if it does so. This is all you need to adjust.

**Comment by rugk** [comment]
Okay, I've adjusted it by myself now. :smile:

**Pull request merged**